root/entity/Collections.js

```diff
@@ -0,0 +1,92 @@
+/*
+ * @flow strict-local
+ * Copyright (C) 2022 MetaBrainz Foundation
+ *
+ * This file is part of MusicBrainz, the open internet music database,
+ * and is licensed under the GPL version 2, or (at your option) any
+ * later version: http://www.gnu.org/licenses/gpl-2.0.txt
+ */
+
+import EditorLink from '../static/scripts/common/components/EditorLink.js';
+import EntityLink from '../static/scripts/common/components/EntityLink.js';
+import chooseLayoutComponent from '../utility/chooseLayoutComponent.js';
+
+type DetailsProps = {
+  +entity: CollectableEntityT,
+  +privateCollectionCount: number,
+  +publicCollections: $ReadOnlyArray<CollectionT>,
+};
+
+const Details = ({
+  entity,
+  publicCollections,
+  privateCollectionCount,
+}: DetailsProps): React$MixedElement => {
+  const entityType = entity.entityType;
+  const LayoutComponent = chooseLayoutComponent(entityType);
+  const publicCollectionCount = publicCollections.length ?? 0;
```

Comment:
Assuming this is always an array, which the Flow type suggests, the length should always be a number.
```suggestion
  const publicCollectionCount = publicCollections.length;
```